### ISSUE TITLE
Wildcard paths working only for collections?

### ISSUE THREAD

**Issue by bugproof** [indice-co/EDI.Net#170]
If I do this:

```cs
var gid = new GID {GoodsItemNumber = 1};
gid.FirstPackagingLevel = new GoodsItemPackageDetails {PackageQuantity=1, PackageTypeDescriptionCode = "201" };
gid.SecondPackagingLevel = new GoodsItemPackageDetails
    {PackageQuantity = 4, PackageTypeDescriptionCode = "201"};
```

then it will serialize as:

GID+1:1:201'
GID+4:201'

but should serialize as

GID+1+1:201+4:201

```cs
[EdiElement]
public class GoodsItemPackageDetails
{
    [EdiValue("9(1)", Path = "GID/*/0", Mandatory = true)] 
    public int PackageQuantity { get; set; }

    [EdiValue("X(17)", Path = "GID/*/1")]
    public string PackageTypeDescriptionCode { get; set; }
}

[EdiSegment(Description = "Goods item details - To indicate totals of a goods item."), EdiPath("GID")]
public class GID
{
    [EdiValue("9(1)", Path = "GID/0")]
    public int GoodsItemNumber { get; set; }
    
    [EdiPath("GID/1")] 
    public GoodsItemPackageDetails FirstPackagingLevel { get; set; }
    
    [EdiPath("GID/2")] 
    public GoodsItemPackageDetails SecondPackagingLevel { get; set; }
    
    [EdiPath("GID/3")] 
    public GoodsItemPackageDetails ThirdPackagingLevel { get; set; }
    
    [EdiPath("GID/4")] 
    public GoodsItemPackageDetails FourthPackagingLevel { get; set; }
    
    [EdiPath("GID/5")] 
    public GoodsItemPackageDetails FifthPackagingLevel { get; set; }
}
```

but if I define GID like:

```cs
[EdiSegment(Description = "Goods item details - To indicate totals of a goods item."), EdiPath("GID")]
public class GID
{
    [EdiValue("9(1)", Path = "GID/0")]
    public int GoodsItemNumber { get; set; }
    
    [EdiPath("GID/*")]
    public List<GoodsItemPackageDetails> PackagingLevels { get; set; } = new List<GoodsItemPackageDetails>(5);
}
```

then everything works fine

What am I doing wrong?

**Comment by cleftheris** [comment]
You are doing everything ok. Seems like a bug.

**Comment by Drenalol** [comment]
@cleftheris Hey!

Can u help please about wildcards with specific index, example:

IFT+1:1+Text1+Text2+Text3

```c#
    [EdiSegment, EdiPath("IFT")]
    public class InteractiveFreeText
    {
        [EdiValue("X(3)", Mandatory = true, Description = "C346-4451", Path = "IFT/0")]
        public string Code { get; set; }

        [EdiValue("X(4)", Description = "C346-9980", Path = "IFT/0/1")]
        public string CodeId { get; set; }
        
        [EdiValue("X(3)", Description = "C346-4405", Path = "IFT/0/2")]
        public string Fare { get; set; }
        
        public List<FreeText> Texts { get; set; }
    }

    [EdiElement, EdiPath("IFT/1") // This take only index [1], but not from [1..*]]
    public class FreeText
    {
        [EdiValue("X(70)", Path = "IFT/*/0")]
        public string Text { get; set; }
    }
```

In indice.EDI is it possible?

**Comment by cleftheris** [comment]
Aha, thanks for the example because these things are not very clear at first glance. 

You have elements which are separated by the plus sign+ and inside them there can be one or more components. There are two ways to discriminate between element types.

1. Discriminate by path 
2. Discriminate by Condition (component value)

In your case you have hard coded that only the elements with Path IFT/1 will go to the list. 
The range notation [1..*] is not implemented and not equivalent to "IFT/1".

I like your range suggestion maybe I can make this work without breaking stuff. I will followup

**Comment by Drenalol** [comment]
Thank you for your investigate, but can u explain me about `Condition` in this example?

```
IFT+4:8+20+SPECIAL SERVICES'

4, 8 - to properties,
20, SPECIAL SERVICES - to List
```

**Comment by cleftheris** [comment]
Can you please take a look at the latest v1.9.13 and this example 
https://github.com/indice-co/EDI.Net/blob/d282d12fcc300eeac57871d82d2fe121cfb83794/test/indice.Edi.Tests/EdiTextReaderTests.cs#L1013
It should work according to your needs. I added the range notation to the element path fragment.

Any feedback is welcome

**Comment by Drenalol** [comment]
Works as expected. Thank you.

`IFT+4:8+20+SPECIAL SERVICES'`

```json
{
  "Code": "4",
  "CodeId": "8",
  "Fare": null,
  "Texts": [
    {
      "Text": "20"
    },
    {
      "Text": "SPECIAL SERVICES"
    }
  ]
}
```

**Comment by Drenalol** [comment]
Hastened with conclusions, if i trying Serialize back to EDI, it will added extra lines

`
UNA:+.? '
IFT+4:8'
IFT+20+SPECIAL SERVICES'
`

**Comment by cleftheris** [comment]
I need to test the serializer more but it is doable. 

Thanks

**Comment by cleftheris** [comment]
Hi @Drenalol please check the latest version (1.9.14). It should resolve the serialization issue.

**Comment by Drenalol** [comment]
@cleftheris Thank you, now deseriazlie & serialize works as expected!

```
IFT+4:8+20+SPECIAL SERVICES'

Deserialize:
{
  "Code": "4",
  "CodeId": "8",
  "Fare": null,
  "Texts": [
    {
      "Text": "20"
    },
    {
      "Text": "SPECIAL SERVICES"
    }
  ]
}

Serialize:
UNA:+.? '
IFT+4:8+20+SPECIAL SERVICES'
```

**Comment by cleftheris** [comment]
@bugproof your original issue should now be resolved in v1.9.16
I got carried away and forgot about it. Please take the time to check it out and close the issue.

C.

**Comment by bugproof** [comment]
@cleftheris I trust you have tested it.
